pandas/core/algorithms.py

```diff
@@ -947,11 +947,26 @@ def select_n_frame(frame, columns, n, method, keep):
     from pandas.core.series import Series
     if not is_list_like(columns):
         columns = [columns]
-    columns = list(columns)
-    ser = getattr(frame[columns[0]], method)(n, keep=keep)
-    if isinstance(ser, Series):
-        ser = ser.to_frame()
-    return ser.merge(frame, on=columns[0], left_index=True)[frame.columns]
+    else:
+        columns = list(columns)
+
+    ascending = method == 'nsmallest'
+    tmp = Series(frame.index)
+    frame.reset_index(inplace=True, drop=True)
```

Comment:
OK fair enough, just in my personal experience I deal with large dataframes (frames that just about fit in memory) and I have to be very careful about copying.

I have to sign off for the day, buy I'll clean this up tomorrow with a copy (also a positive is we only have to copy of the index is duplicated)